### ISSUE TITLE
Deprecate `extension-search-parameters.json` in favor of the `fhir-registry`

### ISSUE THREAD

**Issue by prb112** [IBM/FHIR#1596]
**Is your feature request related to a problem? Please describe.**
Deprecate `extension-search-parameters.json` in favor of the `fhir-registry`

**Describe the solution you'd like**
Currently, customers use `extension-search-parameters` to define search-parameters outside the scope of SearchParameter loaded in Implementation Guides and the Specification.

Note, there are significant updates to tests needed.

**Describe alternatives you've considered**
Keep the existing framework.

**Additional context**
n/a

**Comment by prb112** [comment]
Ran into an issue related to this... https://github.com/IBM/FHIR/issues/2495
Search.init is called before the Registry, however, the registry relies on Search.init to populate the codes for Search, so that the SPs can be processed.  To use this it'd require some rethinking on the initialization and caches that are used.

**Comment by lmsurpre** [comment]
All search parameters (including tenant-specific ones) are now loaded from the registry.
I did *not* formally deprecate the extension-search-parameters.json mechanism...instead I introduced a new RegistryResourceProvider that can read those files and bridge their contents into the registry.

It is possible to use a different method to load tenant-specific search parameters (e.g. using the ServerRegistryResourceProvider), but this was easy enough to continue supporting for now so that we don't needlessly break anyone that may be using this pattern.
